maven-plugin/src/test/java/io/confluent/kafka/schemaregistry/maven/TestCompatibilitySchemaRegistryMojoTest.java

```diff
@@ -0,0 +1,190 @@
+
+import io.confluent.kafka.schemaregistry.client.MockSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import org.apache.avro.Schema;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.hamcrest.core.IsEqual;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class TestCompatibilitySchemaRegistryMojoTest {
+  TestCompatibilitySchemaRegistryMojo mojo;
+  File tempDirectory;
+
+  @Before
+  public void before() throws IOException {
+    this.mojo = new TestCompatibilitySchemaRegistryMojo();
+    this.mojo.client(new MockSchemaRegistryClient());
+    this.tempDirectory = File.createTempFile(this.getClass().getSimpleName(), "tmp");
+    this.tempDirectory.delete();
+    this.tempDirectory.mkdirs();
+  }
+
+  void writeSchema(File outputPath, Schema schema) throws IOException {
+    try (FileWriter writer = new FileWriter(outputPath)) {
+      writer.write(schema.toString(true));
+    }
+  }
+
+  @Test
+  public void register() throws IOException, MojoFailureException, MojoExecutionException, RestClientException {
+    Map<String, Boolean> expectedVersions = new LinkedHashMap<>();
+
+    Map<String, File> subjectToFile = new LinkedHashMap<>();
+    for (int i = 0; i < 100; i++) {
+      String keySubject = String.format("TestSubject%03d-Key", i);
+      String valueSubject = String.format("TestSubject%03d-Value", i);
+      Schema keySchema = Schema.create(Schema.Type.STRING);
+      Schema valueSchema = Schema.createUnion(Arrays.asList(Schema.create(Schema.Type.STRING), Schema.create(Schema.Type.NULL)));
+      this.mojo.client().register(keySubject, keySchema);
+      this.mojo.client().register(valueSubject, valueSchema);
+      File keySchemaFile = new File(this.tempDirectory, keySubject + ".avsc");
+      File valueSchemaFile = new File(this.tempDirectory, valueSubject + ".avsc");
+      writeSchema(keySchemaFile, keySchema);
+      writeSchema(valueSchemaFile, valueSchema);
+      subjectToFile.put(keySubject, keySchemaFile);
+      expectedVersions.put(keySubject, true);
+      subjectToFile.put(valueSubject, valueSchemaFile);
+      expectedVersions.put(valueSubject, true);
+    }
+
+    this.mojo.subjects = subjectToFile;
+    this.mojo.execute();
+
+    Assert.assertThat(this.mojo.schemaCompatibility, IsEqual.equalTo(expectedVersions));
+  }
+
+  void writeMalformedFile(File file) throws IOException {
+    try (FileWriter writer = new FileWriter(file)) {
+      writer.write("[");
+    }
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void malformedSchema() throws IOException, MojoFailureException, MojoExecutionException, RestClientException {
```

Comment:
Seems like the malformed and missing schema tests could probably just be handled by a test of `SchemaRegistryMojo` since it is shared functionality in `loadSchemas`.